File: backend/test_drag_drop.py
import httpx


# Test drag-drop functionality
def test_drag_drop():
    client = httpx.Client(base_url="http://localhost:8000")

    # Create a board
    board = client.post("/api/boards/", json={"name": "Drag Drop Test"}).json()
    board_id = board["id"]
    print(f"✅ Created board: {board['name']} (ID: {board_id})")

    # Create a ticket
    ticket = client.post(
        "/api/tickets/",
        json={"title": "Drag Test Ticket", "board_id": board_id, "current_column": "todo"},
    ).json()
    ticket_id = ticket["id"]
    print(
        f"✅ Created ticket: {ticket['title']} (ID: {ticket_id}) "
        f"in column: {ticket['current_column']}"
    )

    # Move the ticket (drag-drop simulation) - using POST
    move_response = client.post(
        f"/api/tickets/{ticket_id}/move", json={"column": "in_progress", "moved_by": "test_user"}
    )

    if move_response.status_code == 200:
        updated_ticket = move_response.json()
        print(f"✅ Moved ticket to: {updated_ticket['current_column']}")
        print("   Broadcast info: WebSocket event 'ticket_moved' sent to all clients")
    else:
        print(f"❌ Failed to move ticket: {move_response.text}")

    # Verify the move
    verify = client.get(f"/api/tickets/{ticket_id}").json()
    if verify["current_column"] == "in_progress":
        print(f"✅ Verified: Ticket is now in '{verify['current_column']}' column")
    else:
        print(f"❌ Verification failed: Ticket is in '{verify['current_column']}' column")

    # Test another move
    move2 = client.post(
        f"/api/tickets/{ticket_id}/move", json={"column": "done", "moved_by": "test_user"}
    )
    if move2.status_code == 200:
        print("✅ Moved ticket to: done")

    # Cleanup
    client.delete(f"/api/boards/{board_id}")
    print("🧹 Cleaned up test data")

    return True


if __name__ == "__main__":
    print("=== Drag-Drop Functionality Test ===\n")
    try:
        test_drag_drop()
        print("\n✅ All drag-drop tests passed\\!")
    except Exception as e:
        print(f"\n❌ Test failed: {e}")

File: frontend/test-drag-drop.js
// Test script to verify drag and drop API functionality
const API_BASE = 'http://localhost:18000';

async function testDragDrop() {
  console.log('Testing drag and drop API...\n');

  try {
    // First, get boards
    const boardsResponse = await fetch(`${API_BASE}/api/boards/`);
    const boards = await boardsResponse.json();

    if (!boards || boards.length === 0) {
      console.log('No boards found. Please create a board first.');
      return;
    }

    const boardId = boards[0].id;
    console.log(`Using board ID: ${boardId}`);

    // Get tickets in the board
    const ticketsResponse = await fetch(`${API_BASE}/api/tickets/?board_id=${boardId}`);
    const ticketsData = await ticketsResponse.json();
    const tickets = ticketsData.items || [];

    if (tickets.length === 0) {
      console.log('No tickets found. Creating a test ticket...');
      const createResponse = await fetch(`${API_BASE}/api/tickets/`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          title: 'Test Drag-Drop Card',
          description: 'This card will test drag and drop',
          board_id: boardId,
          current_column: 'Not Started'
        })
      });
      const newTicket = await createResponse.json();
      tickets.push(newTicket);
    }

    const testTicket = tickets[0];
    console.log(`\nTest ticket: "${testTicket.title}" (ID: ${testTicket.id})`);
    console.log(`Current column: ${testTicket.current_column}`);

    // Test moving the ticket to different columns
    const columns = ['In Progress', 'Blocked', 'Ready for QC', 'Done', 'Not Started'];
    const targetColumn = columns.find(col => col !== testTicket.current_column) || 'In Progress';

    console.log(`\nMoving ticket to "${targetColumn}"...`);

    const moveResponse = await fetch(`${API_BASE}/api/tickets/${testTicket.id}/move`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        column: targetColumn
      })
    });

    if (!moveResponse.ok) {
      const error = await moveResponse.text();
      throw new Error(`Failed to move ticket: ${moveResponse.status} - ${error}`);
    }

    const movedTicket = await moveResponse.json();
    console.log('\n✅ Ticket moved successfully!');
    console.log(`New column: ${movedTicket.current_column}`);

    // Verify the ticket is in the new column
    const verifyResponse = await fetch(`${API_BASE}/api/tickets/${testTicket.id}`);
    const verifiedTicket = await verifyResponse.json();

    if (verifiedTicket.current_column === targetColumn) {
      console.log('\n✅ Move verified! Ticket is in the correct column.');
    } else {
      console.log(`\n⚠️ Warning: Ticket column mismatch. Expected: ${targetColumn}, Got: ${verifiedTicket.current_column}`);
    }

    // Test moving back to original column
    console.log(`\nMoving ticket back to "${testTicket.current_column}"...`);
    const moveBackResponse = await fetch(`${API_BASE}/api/tickets/${testTicket.id}/move`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        column: testTicket.current_column
      })
    });

    if (moveBackResponse.ok) {
      console.log('✅ Ticket moved back successfully!');
    }

  } catch (error) {
    console.error('\n❌ Test failed:', error.message);
    process.exit(1);
  }
}

// Run the test
testDragDrop().then(() => {
  console.log('\n✅ All drag-drop tests passed!');
  process.exit(0);
});
